data/src/main/java/com/linkedin/data/template/DynamicRecordTemplate.java

```diff
@@ -269,11 +271,11 @@ private <T> void putArray(RecordDataSchema.Field field, FieldDef<T> fieldDef, T
       {
         data.add(
             DataTemplateUtil.coerceInput(item,
-                                         (Class<Object>)item.getClass(),
-                                         itemType.isEnum() ? String.class : itemType));
+                (Class<Object>)item.getClass(),
+                itemType.isEnum() ? String.class : itemType));
       }
     }
 
-    putDirect(field, DataList.class, data, SetMode.DISALLOW_NULL);
+    CheckedUtil.putWithoutChecking(_map, field.getName(), data);
```

Comment:
why not use `unsafePutWrapped`?

Reply:
DataList is not wrapped. putDirect causes an unnecessary coerced call.